polaris-react/src/components/Text/Text.tsx

```diff
@@ -30,11 +30,13 @@ type Variant =
 
 type Alignment = 'start' | 'center' | 'end' | 'justify';
 
-type FontWeight = 'regular' | 'medium' | 'semibold' | 'bold';
+type FontWeight = 'regular' | 'medium' | 'semibold' | 'bold' | null;
 
 type Color = 'success' | 'critical' | 'warning' | 'subdued' | 'text-inverse';
 
-const VariantFontWeightMapping: {[V in Variant]: FontWeight} = {
+const VariantFontWeightMapping: {
+  [V in Variant]: FontWeight;
```

Comment:
```suggestion
  [V in Variant]: FontWeight | undefined;
```

Can we add `undefined` here and remove `null` from the `FontWeight` union? This update should satisfy the original types and the new behavior.